torch/_inductor/pattern_matcher.py

```diff
@@ -1580,7 +1580,7 @@ def gen_pattern(
     search_gm = trace_fn(search_fn, flat_inputs)
     return fx_to_pattern(
         search_gm,
-        ignore_types=(int, float, list, torch.device, torch.dtype),
+        ignore_types=(int, float, list, torch.device),
```

Comment:
If ignore_types contains `torch.dtype`, the fwd (dtype=e4m3) and bwd (dtype=e5m2) patterns will fail at `assert pattern_repr not in _seen_patterns`, https://github.com/pytorch/pytorch/blob/main/torch/_inductor/pattern_matcher.py#L1398

Reply:
Don't think we can land this but https://github.com/pytorch/pytorch/pull/139321 should fix

Reply:
thanks for the comment. I'll remove this change and test the pr again with https://github.com/pytorch/pytorch/pull/139321

Reply:
landed ! we could allow register_replacement api to specify an override of `ignore_types` - just not make a global change like here. in any case, with the pr, shouldn't need any further change.

Reply:
It works with your pr. thanks!